addons/website/controllers/main.py

```diff
@@ -641,6 +642,40 @@ def pagenew(self, path="", add_menu=False, template=False, redirect=False, **kwa
             return json.dumps({'view_id': page.get('view_id')})
         return json.dumps({'url': url})
 
+    @http.route('/website/new_page_templates', type='json', auth='user', website=True)
+    def new_page_templates(self, **kw_args):
+        View = request.env['ir.ui.view']
+        result = []
+        groups_el = etree.fromstring(View._render_template('website.new_page_template_groups'))
+        for group_el in groups_el.getchildren():
+            templates = []
+            group = {
+                'id': group_el.attrib['id'],
+                'title': group_el.text,
+                'templates': templates,
+            }
+
+            def adjust_template(template_html):
+                tree = html.fromstring(template_html)
+                View._adjust_new_page_template(tree)
```

Comment:
Why this is a method, let alone on `ir.ui.view`? Can't this be inlined here? It's only used once

Reply:
The purpose was originally to make it overridable by e.g. `website_sale` to adjust the products snippet - but I ultimately introduced `_prepare_new_page_template_context` for that, so I think I can remove it indeed.